src/main/java/athena/task/Task.java

```diff
@@ -71,22 +75,89 @@ public Task(String name, String startTime, String duration, String deadline,
+            LocalDate wednesdayDate = getFirstDay(DayOfWeek.WEDNESDAY);
+            addDates(wednesdayDate);
+        } else if (recurrence.toUpperCase().equals(Recurrence.THURSDAY.toString())) {
+            LocalDate thursdayDate = getFirstDay(DayOfWeek.THURSDAY);
+            addDates(thursdayDate);
+        } else if (recurrence.toUpperCase().equals(Recurrence.FRIDAY.toString())) {
+            LocalDate fridayDate = getFirstDay(DayOfWeek.FRIDAY);
+            addDates(fridayDate);
+        } else if (recurrence.toUpperCase().equals(Recurrence.SATURDAY.toString())) {
+            LocalDate saturdayDate = getFirstDay(DayOfWeek.SATURDAY);
+            addDates(saturdayDate);
+        } else if (recurrence.toUpperCase().equals(Recurrence.SUNDAY.toString())) {
+            LocalDate sundayDate = getFirstDay(DayOfWeek.SUNDAY);
+            addDates(sundayDate);
         } else {
             try {
-                DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd-MM-yyyy");
-                recurrenceDate = LocalDate.parse(recurrence, formatter);
+                setRecurrenceDate(recurrence);
             } catch (DateTimeParseException e) {
                 // TODO: Handle this properly
                 System.out.println("I don't understand the date you gave. So I set it to today.");
-                recurrenceDate = LocalDate.now();
+                recurrenceDates.add(LocalDate.now());
             }
         }
     }
 
+    private void addDates(LocalDate startDate) {
+        for (int i = 0; i < 10; i++) { // Max number of weeks now is 10 to prevent infinite recurrence
+            recurrenceDates.add(startDate.plusWeeks(i));
+        }
+    }
+
+    private LocalDate getFirstDay(DayOfWeek dayOfWeek) {
+        LocalDate startDate = LocalDate.now();
+        for (int i = 1; i < 7; i++) {
+            if (startDate.getDayOfWeek().equals(dayOfWeek)) {
+                break;
+            } else {
+                startDate = startDate.plusDays(1);
+            }
+        }
+        return startDate;
+    }
+
+    private void setRecurrenceDate(String recurrence) {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd-MM-yyyy");
+        if (recurrence.length() == DATE_TIME_FORMAT) {
+            String year = Integer.toString(LocalDate.now().getYear());
+            this.recurrence = recurrence + "-" + year;
+            recurrenceDates.add(LocalDate.parse(recurrence + "-" + year, formatter));
+        } else {
+            recurrenceDates.add(LocalDate.parse(recurrence, formatter));
+        }
+    }
+
+    public LocalDate getRecurrenceDate(String date) {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+        return LocalDate.parse(date, formatter);
+    }
+
+    public Task getClone() {
+        Task copy = new Task(name, startTime, duration, deadline, recurrence,
+                importance, notes, number, isFlexible);
```

Comment:
not sure if we should support having tasks with the same task number, as long as the cloned tasks are not saved should be ok.